docker/docker-entrypoint.sh

```diff
@@ -1,28 +1,39 @@
 #!/bin/bash
 
 # Parameters
-if [[ "${NAT_IP_AUTO}" == "true" && -z "${NAT_PUBLIC_IP_AUTO}" ]]; then
-  export CODEX_NAT=$(hostname --ip-address)
-  echo "Internal: Set CODEX_NAT: ${CODEX_NAT}"
+if [[ -z "${CODEX_NAT}" ]]; then
+  if [[ "${NAT_IP_AUTO}" == "true" && -z "${NAT_PUBLIC_IP_AUTO}" ]]; then
+    export CODEX_NAT=$(hostname --ip-address)
+    echo "Private: CODEX_NAT=${CODEX_NAT}"
+  elif [[ -n "${NAT_PUBLIC_IP_AUTO}" ]]; then
+    # Run for 60 seconds if fail
+    WAIT=120
+    SECONDS=0
+    SLEEP=5
```

Comment:
Good question, we may changes that and just to consider some points
- for curl we have `-m 5` - 5 seconds timeout for IP Lookup service
- `sleep 5` - 5 seconds before a new lookup try

If IP Lookup service works fine and we get Public IP - start time will not be affected (~ 1 sec), but in case of the issue, it make sense to wait
- temporary network issue - short try make sense
- service overload - short try make no much sense
- relate limit - short try make no much sense